core/src/tangram.cpp

```diff
@@ -119,33 +138,105 @@ void setScene(std::shared_ptr<Scene>& _scene) {
 void loadScene(const char* _scenePath) {
     LOG("Loading scene file: %s", _scenePath);
 
-    auto sceneString = stringFromFile(setResourceRoot(_scenePath).c_str(), PathType::resource);
-
-    // Copy old scene
-    auto scene = std::make_shared<Scene>(*m_scene);
-
-    if (SceneLoader::loadScene(sceneString, *scene)) {
-        setScene(scene);
+    {
+        std::lock_guard<std::mutex> lock(m_newScenePathMutex);
+        m_newScenePath = std::string(_scenePath);
+        m_newScene = true;
     }
+
+    if (m_sceneLoading) { return; }
+
+    m_sceneLoading = true;
+
+    std::thread t([&]() {
+            while(m_newScene) {
+                std::string scenePath;
+                {
+                    std::lock_guard<std::mutex> lock(m_newScenePathMutex);
+                    scenePath = setResourceRoot(m_newScenePath.c_str());
+                    m_newScene = false;
+                }
+
+                // Copy old scene
+                std::shared_ptr<Scene> scene;
+                {
+                    std::lock_guard<std::mutex> lock(m_sceneMutex);
+                    scene = std::make_shared<Scene>(*m_scene);
+                }
+
```

Comment:
The view was copied when loading a new scene so that if loading the new scene failed, the view could be returned to its previous, valid configuration.